### ISSUE TITLE
MS Windows clamav-0.104.1.win.x64.msi broke freshclam.exe and clamscan.exe - .dll errors

### ISSUE THREAD

**Issue by caganer** [Cisco-Talos/clamav#371]
(A) After noting the message in log output of freshclam.exe I upgraded from preceding version of ClamAV to clamav-0.104.1.win.x64.msi on Windows 7 Professional SP1 which has broken at least freshclam.exe and clamscan.exe  

Error message splash screen on running freshclam.exe: 
"The procedure entry point cli_strdup could not be located in the dynamic link library libclamav.dll."

This occurs both where (i) the msi installation is run without following the steps (formerly) at 
https://www.clamav.net/documents/installing-clamav-on-windows [ie settings made using Powershell - see below *]
and
(ii) after doing the settings procedure in powershell as described above AND after rebooting.

(B) Likewise after the aforesaid steps, running clamscan.exe results in error screen: 
"The procedure entry point cli_sigperf_print could not be located in the dynamic link library libclamav.dll."

[*] I note the above Windows installation guide now appears to be at:
https://docs.clamav.net/manual/Usage/Configuration.html#windows

**Comment by micahsnyder** [comment]
Can you confirm that you fully removed the old version before installing 0.104.1?  The error you're describing sounds to me like you may have an older clamav DLL being used by the newer freshclam.exe (or maybe the other way around).

**Comment by markpizz** [comment]
I'm not using the .msi installer, however a more robust Windows installer would detect the prior installation and usually prompt about removing it or just remove it automatically before installing the newer version.

I use the .zip file and had no problems dumping the contained .dlls and .exe on top of the prior versions after stopping the running freshclam and clamd processes.

**Comment by caganer** [comment]
I just downloaded the *.zip version of the (installed, non-working) latest 
*.msi version and copied over its content to the installation directory 
checking that the *.conf files stayed good and have got freshclam and 
scanclam to work so hopefully that's that.

I can't recall for sure now but I am fairly certain I just ran the *.msi 
without uninstalling the previous version first - because the following 
doesn't tell you to uninstal first:

https://docs.clamav.net/manual/Usage/Configuration.html#windows

- unless I've missed it in all the detail 'LOL'...

Regards


Robert Jones




On Tue, 16 Nov 2021, Mark Pizzolato wrote:

> Date: Tue, 16 Nov 2021 13:40:56 -0800
> From: Mark Pizzolato ***@***.***>
> Reply-To: Cisco-Talos/clamav
>     ***@***.***>
> To: Cisco-Talos/clamav ***@***.***>
> Cc: caganer ***@***.***>, Author ***@***.***>
> Subject: Re: [Cisco-Talos/clamav] MS Windows clamav-0.104.1.win.x64.msi broke
>     freshclam.exe and clamscan.exe - .dll errors (Issue #371)
>

> I'm not using the .msi installer, however a more robust Windows 
> installer would detect the prior installation and usually prompt about 
> removing it or just remove it automatically before installing the newer 
> version.
>
> I use the .zip file and had no problems dumping the contained .dlls and 
> .exe on top of the prior versions after stopping the running freshclam 
> and clamd processes.

>
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Cisco-Talos/clamav/issues/371#issuecomment-970703045

THIS E-MAIL AND ANY ATTACHED FILES ARE CONFIDENTIAL AND MAY BE LEGALLY
PRIVILEGED. If you are not the addressee, any disclosure,
reproduction, copying, distribution or other dissemination or use of
this communication is strictly prohibited. If you have received this
transmission in error please notify the sender immediately and then
delete this e-mail. All liability for viruses is excluded.

***@***.***
***@***.***)
SDF Public Access UNIX System - http://SDF.org

**Comment by micahsnyder** [comment]
> I'm not using the .msi installer, however a more robust Windows installer would detect the prior installation and usually prompt about removing it or just remove it automatically before installing the newer version.
> 
> I use the .zip file and had no problems dumping the contained .dlls and .exe on top of the prior versions after stopping the running freshclam and clamd processes.

The tooling we use to make ClamAV's installer has some features to make upgrades smoother, but only in 0.104.0+ and it may not be configured correctly.  It's newer and we haven't done a lot of testing to make sure it's all set correctly.

@caganer I'm glad to see that you resolved it in the end.  I did try testing the `clamav-0.104.1.win.x64.msi` installer on a windows 7 x64 Enterprise SP1 VM that's mostly clean.  I uninstalled previously installed VC runtime packages to make sure that the clamav installer provides all the required DLL's.  It worked okay for me.  Both clamscan and freshclam ran as expected.  

I'm not sure how your install got into a bad state, but glad to know you have a working install now.
